Modules/Core/Common/include/itkShapedImageNeighborhoodRange.h

```diff
@@ -598,36 +599,39 @@ class ShapedImageNeighborhoodRange final
 
   // ShapedImageNeighborhoodRange data members (strictly private):
 
-  // Pointer to the buffer of the image. Should not be null.
-  QualifiedInternalPixelType* m_ImageBufferPointer;
+  // Pointer to the buffer of the image.
+  QualifiedInternalPixelType* m_ImageBufferPointer{ nullptr };
 
   // Index and size of the buffered image region.
-  RegionData m_BufferedRegionData;
+  RegionData m_BufferedRegionData{};
 
   // A copy of the offset table of the image.
-  OffsetType m_OffsetTable;
+  OffsetType m_OffsetTable{{}};
 
-  NeighborhoodAccessorFunctorType m_NeighborhoodAccessor;
+  NeighborhoodAccessorFunctorType m_NeighborhoodAccessor{};
 
   // Index (pixel coordinates) of the location of the neighborhood relative
   // to the origin of the image. Typically it is the location of the
   // center pixel of the neighborhood. It may be outside the image boundaries.
-  IndexType m_RelativeLocation;
+  IndexType m_RelativeLocation{{}};
 
   // The offsets relative to m_RelativeLocation that specify the neighborhood shape.
-  const OffsetType* m_ShapeOffsets;
+  const OffsetType* m_ShapeOffsets{ nullptr };
 
   // The number of neighborhood pixels.
-  const std::size_t m_NumberOfNeighborhoodPixels;
+  const std::size_t m_NumberOfNeighborhoodPixels{ 0 };
 
-  const OptionalPixelAccessParameterType m_OptionalPixelAccessParameter;
+  const OptionalPixelAccessParameterType m_OptionalPixelAccessParameter{};
 
 public:
   using const_iterator = QualifiedIterator<true>;
   using iterator = QualifiedIterator<false>;
   using reverse_iterator = std::reverse_iterator<iterator>;
   using const_reverse_iterator = std::reverse_iterator<const_iterator>;
 
+  /** Default-constructor. Constructs an empty range. */
+  ShapedImageNeighborhoodRange() = default;
```

Comment:
You don't want copy-constructor, move-constructor, assignment operator? Perhaps follow rule of 5 or rule of zero. Or explicitly delete the others.